component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -1557,4 +1569,26 @@ protected boolean retryAuthenticationEnabled() {
 
         return configProperties;
     }
-}
\ No newline at end of file
+
+
+    protected void triggerEvent(String userName, String tenantDomain, String userStoreDomainName,
+                                String notificationEvent, String otpCode) {
+
+        String eventName = IdentityEventConstants.Event.TRIGGER_NOTIFICATION;
+
+        HashMap<String, Object> properties = new HashMap<>();
+        properties.put(IdentityEventConstants.EventProperty.USER_NAME, userName);
+        properties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, userStoreDomainName);
+        properties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, tenantDomain);
+        properties.put(EmailOTPAuthenticatorConstants.CODE, otpCode);
+
+        properties.put(EmailOTPAuthenticatorConstants.TEMPLATE_TYPE, notificationEvent);
+        Event identityMgtEvent = new Event(eventName, properties);
+        try {
+            EmailOTPServiceDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+        } catch (Exception e) {
```

Comment:
Use the specific exception (IdentityEventException) instead of the common exception.

Reply:
Added in https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/26/commits/4c8a1a700c541009035ba290999b9609af6e95d2